### PR TITLE
[libc++][chrono] Remove non-standard relational operators for `std::chrono::weekday`

### PR DESCRIPTION
**Pull request by frederick-vs-ja**
These operators are absent in [[time.syn]](https://eel.is/c++draft/time.syn), and a note in [[time.cal.wd.overview]/1](https://eel.is/c++draft/time.cal.wd.overview#1) indicates that the absence is intended.

This patch removes the undocumented extension, while providing a migration path for vendors by providing the `_LIBCPP_ENABLE_REMOVED_WEEKDAY_RELATIONAL_OPERATORS` macro. This macro will be honored for the LLVM 19 release and will be removed after that, at which point allocator will be removed unconditionally.

### CODE REVIEW

File: libcxx/docs/ReleaseNotes/19.rst [hunk 1]
```diff
@@ -134,6 +134,12 @@ Deprecations and Removals
   `std-allocator-const <https://clang.llvm.org/extra/clang-tidy/checks/portability/std-allocator-const.html>`
   enabled.
 
+- libc++ no longer supports relational comparison for ``std::chrono::weekday``. The relational comparison operators were
+  provided as an undocumented extension. If you were using relational comparison on ``std::chrono::weekday``, compare
+  the results of ``c_encoding()`` or ``iso_encoding()`` instead. The
+  ``_LIBCPP_ENABLE_REMOVED_WEEKDAY_RELATIONAL_OPERATORS`` macro can be defined to temporarily re-enable this extension
+  as folks transition their code. This macro will be honored for one released and ignored starting in LLVM 20.
```
Comment (mordante) [inline libcxx/docs/ReleaseNotes/19.rst:141]: Let's remove the folks part, it does not fit the style of the libc++ release notes.
```suggestion
  ``_LIBCPP_ENABLE_REMOVED_WEEKDAY_RELATIONAL_OPERATORS`` macro can be defined to temporarily re-enable this extension.
  This macro will be honored for one released and ignored starting in LLVM 20.
```

File: libcxx/test/std/time/time.cal/time.cal.weekday/time.cal.weekday.nonmembers/comparisons.pass.cpp [hunk 2]
```diff
@@ -17,10 +17,13 @@
 #include <chrono>
 #include <type_traits>
 #include <cassert>
+#include <concepts>
 
 #include "test_macros.h"
 #include "test_comparisons.h"
 
+static_assert(!std::totally_ordered<std::chrono::weekday>);
```
Comment (mordante) [inline libcxx/test/std/time/time.cal/time.cal.weekday/time.cal.weekday.nonmembers/comparisons.pass.cpp:25]: Interesting the relational operations are never tested.

**Review by mordante — APPROVED** [review]
Nice catch! LGTM modulo a minor comment.

File: libcxx/docs/ReleaseNotes/19.rst [hunk 3]
```diff
@@ -134,6 +134,12 @@ Deprecations and Removals
   `std-allocator-const <https://clang.llvm.org/extra/clang-tidy/checks/portability/std-allocator-const.html>`
   enabled.
 
+- libc++ no longer supports relational comparison for ``std::chrono::weekday``. The relational comparison operators were
+  provided as an undocumented extension. If you were using relational comparison on ``std::chrono::weekday``, compare
+  the results of ``c_encoding()`` or ``iso_encoding()`` instead. The
+  ``_LIBCPP_ENABLE_REMOVED_WEEKDAY_RELATIONAL_OPERATORS`` macro can be defined to temporarily re-enable this extension.
+  This macro will be honored for one released and ignored starting in LLVM 20.
```
Comment (h-vetinari) [inline libcxx/docs/ReleaseNotes/19.rst:141]: Typo
```suggestion
  This macro will be honored for one release and ignored starting in LLVM 20.
```

File: libcxx/docs/ReleaseNotes/19.rst [hunk 4]
```diff
@@ -134,6 +134,12 @@ Deprecations and Removals
   `std-allocator-const <https://clang.llvm.org/extra/clang-tidy/checks/portability/std-allocator-const.html>`
   enabled.
 
+- libc++ no longer supports relational comparison for ``std::chrono::weekday``. The relational comparison operators were
+  provided as an undocumented extension. If you were using relational comparison on ``std::chrono::weekday``, compare
+  the results of ``c_encoding()`` or ``iso_encoding()`` instead. The
+  ``_LIBCPP_ENABLE_REMOVED_WEEKDAY_RELATIONAL_OPERATORS`` macro can be defined to temporarily re-enable this extension.
+  This macro will be honored for one release and ignored starting in LLVM 20.
+
```
Comment (ldionne) [inline libcxx/docs/ReleaseNotes/19.rst:142]: Can you also add an entry in "Upcoming deprecations and removals" for LLVM 20 saying that the macro will be removed? This mostly acts as a reminder for us when we start working on LLVM 20 -- we go through the upcoming deprecations/removals from the previous release and make sure they happen.

**Review by ldionne — COMMENTED** [review]
Good find, thanks for the patch!

File: libcxx/docs/ReleaseNotes/19.rst (hunk 4 above)
Comment (frederick-vs-ja) [reply libcxx/docs/ReleaseNotes/19.rst:149]: Done. It seems that `20.rst` should also be updated so I also added an entry inside it.

**Review by ldionne — APPROVED** [review]

**Pull request merged**